doc/source/Interface/ControlsAndShortcuts/generate.sh

```diff
@@ -0,0 +1,9 @@
+#! /bin/bash
+# BuildTarget: images/M1.png
+
+set -e
+
+cp $GAFFER_ROOT/doc/gaffer/graphics/M1.png images
+cp $GAFFER_ROOT/doc/gaffer/graphics/M2.png images
+cp $GAFFER_ROOT/doc/gaffer/graphics/M3.png images
+cp $GAFFER_ROOT/doc/gaffer/graphics/MW.png images
```

Comment:
Yes, it's possible if we write the images to `_static` first.

Pros:
- No need for the `gaffer/doc/gaffer/graphics` directory, `generate.sh`, or`.gitignore` entries.

Cons:
- Contributors need to determine ahead of time the doc's relative path to `_static`, requiring a bit more work when writing the `.md` file.
- Inconsistency: we already redundantly copy UI icons from `gaffer/graphics` in the various `generate.sh` files. If we change how we do it here, to make the process consistent we should copy all of `gaffer/graphics` to `gaffer/doc/gaffer/html/_static` too.

For that matter, for complete consistency, we could make **all** images go to `_static` – the validator treats them the same. This is tempting, as it would solve the issue of the screengrabs dirtying the source before being copied to the build.

But I think this is all out of scope. Let's keep it as-is and revisit this improvement later.

Reply:
I think there is a difference here as `gaffer/graphics` is part of the application distribution, so needs copying out so the docs can be hosted on the website. These new images are _solely_ for docs though, and we can building them directly into somewhere that is part of the docs tree. This means they will already be part of what is hosted on the website. Is there no way in the `.. |M1| image:: images/M1.png` syntax to specify an absolute path to the root of the docs build rather than a relative one?

It is somewhat in scope as we're adding a new mechanism here (docs specific image generation), so the comment isn't so much about "we have this approach of copying Gaffer images already, can we change it" (which is totally out of scope) as "do we really need to be using that for these new images - is it the right thing to be introducing here"?

Reply:
> `gaffer/graphics` is part of the application distribution, so needs copying out so the docs can be hosted on the website

Yes, but we copy them into [**each**](https://github.com/GafferHQ/gaffer/blob/master/doc/source/WorkingWithTheNodeGraph/TutorialUsingTheOSLCodeNode/generate.sh#L6) [**respective**](https://github.com/GafferHQ/gaffer/blob/master/doc/source/WorkingWithScenes/Camera/generate.sh#L11) [**article**](https://github.com/GafferHQ/gaffer/blob/master/doc/source/GettingStarted/TutorialAssemblingTheGafferBot/generate.sh#L7) that uses them, as needed. I'm not saying it's ideal, but it's what happens.

> we can building them directly into somewhere that is part of the docs tree

One could make this argument about every image in the docs. But we use the same method ubiquitously, regardless of how the image is made or whether it should be single-sourced. I'm currently building and compositing the automated screengrabs – images meant only for the docs – into the target article's relative `images` directory, for consistency's sake. It is in fact problematic from a pollution standpoint, and I'd like to move away from this structure entirely at some point.

As it stands, a contributor will expect one location for an article's images: the relative `images` directory. This is the case for every instance of an image in the docs, with the exception of the logo on the home page, which I linked to directly as a hack.